org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/M2EUtils.java

```diff
@@ -61,7 +61,8 @@ public static void createFolder(IFolder folder, boolean derived, IProgressMonito
         }
       } catch(CoreException ex) {
         //Don't fail if the resource already exists, in case of a race condition
-        if(ex.getStatus().getCode() != IResourceStatus.RESOURCE_EXISTS) {
+        int code = ex.getStatus().getCode();
+        if(code != IResourceStatus.RESOURCE_EXISTS && code != IResourceStatus.PATH_OCCUPIED) {
           throw ex;
         }
```

Comment:
Instead of checking the error code, would it be possible to check resource existence
```suggestion
        if (!folder.exists()) {
          throw ex;
        }
```

Reply:
The only reason I see not do that would be the performance cost, but it seems rather marginal here, so I'm good with that.